odoo/addons/base/models/ir_http.py

```diff
@@ -280,7 +280,8 @@ def get_translations_for_webclient(self, modules, lang):
                 "time_format": langs.time_format,
                 "grouping": langs.grouping,
                 "decimal_point": langs.decimal_point,
-                "thousands_sep": langs.thousands_sep,
+                # A0 is the unicode codepoint for a non-breaking space
+                "thousands_sep": langs.thousands_sep.replace(' ', '\xa0'),
```

Comment:
```suggestion
                "thousands_sep": (langs.thousands_sep or '').replace(' ', '\xa0'),
```
Need to be backport if accepted